### DISCUSSION TITLE
pinning of PHP

### DISCUSSION THREAD

**Discussion by florisluiten** [renovatebot/renovate#18686] [Suggest an Idea]
Hi there,

We have used renovate since a while, and notices a new behavior since a couple of days.

Take the following composer.json:

```json
{
    "name": "what/ever",
    "description": "Whatever",
    "license": "proprietary",
    "require": {
        "php": "^8.1"
    }
}
```

And the following config.js:

```js
module.exports = {
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "rangeStrategy": "pin"
};
```

Previously, this did not cause a pr to be opened to pin "php", since it was detected as a "system requirement", and not as a package. This makes sense, since composer cannot manage the installed PHP version anyways.
However, after upgrading to the latest version, I noticed that pr's were opened to pin PHP to a specific version, like so:

```diff
    "name": "what/ever",
    "description": "Whatever",
    "require": {
-        "php": "^8.1",
+        "php": "8.1.12",
    }
}
```

This is not what I want though, because that would require me to install the exact same version of PHP on every machine; and it didn't create these pr's previously. So to me, this seems like a bug.

I've dugged into the code, and found out that commit 0b2a14e3a2c52 introduced this

I'm starting this discussion to see if more agree that this is a bug that needs fixing, or that it is expected behavior.

**Comment by rarkins** [comment]
It's expected behaviour if you have set a blanket rangeStrategy=pin config

**Reply by florisluiten** [reply]
Ok, thanks for your response :)
